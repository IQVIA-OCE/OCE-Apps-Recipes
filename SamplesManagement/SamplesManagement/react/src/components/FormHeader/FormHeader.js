import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { Button, themeGrey, withTheme, Text } from '@oce-apps/apollo-react-native';
import color from 'color';

class FormHeader extends Component {
  render() {
    const {
      icon,
      iconColor,
      label,
      title,
      controls,
      children,
      theme,
    } = this.props;
    const iconName = icon ? icon : 'account';
    let borderColor = themeGrey[200];
    let labelColor = color(theme.colors.text).darken(0.85).hex();

    if (theme.dark) {
      borderColor = "none";
      labelColor = color(theme.colors.text).darken(0.12).hex();
    }

    return (
      <View
        style={[styles.container, { backgroundColor: theme.colors.background, borderColor }]}
      >
        <View
          style={[
            styles.icon,
            {
              backgroundColor: iconColor ? iconColor : themeGrey[600],
            },
          ]}
        >
          <Icon name={iconName} size={32} color={'#fff'} />
        </View>
        <View style={styles.textContainer}>
          {label && (
            <Text style={[styles.label, { color: labelColor }]}>
              {label.toUpperCase()}
            </Text>
          )}
          <Text style={styles.title}>{title}</Text>
        </View>
        <View>{children}</View>
        <View style={styles.buttonsContainer}>
          {controls &&
            controls.map(({ label, color, mode, onPress, disabled }, i) => {
              return (
                <Button
                  key={i}
                  onPress={onPress}
                  size="small"
                  mode={mode ? mode : 'outlined'}
                  color={color ? color : 'secondary'}
                  style={styles.button}
                  disabled={disabled}
                >
                  {label}
                </Button>
              );
            })}
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderWidth: 1,
    padding: 12,
    alignItems: 'center',
  },
  icon: {
    borderRadius: 3,
    marginRight: 10,
    height: 35,
    width: 35,
    alignItems: 'center',
  },
  textContainer: {
    flexGrow: 1,
  },
  label: {
    letterSpacing: 2,
    fontSize: 12,
  },
  title: {
    fontWeight: 'bold',
    fontSize: 17,
  },
  buttonsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 4,
  },
  button: {
    marginLeft: 7,
  },
});

export default withTheme(FormHeader);
