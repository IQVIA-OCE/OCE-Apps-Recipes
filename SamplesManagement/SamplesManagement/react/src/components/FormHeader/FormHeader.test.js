import React from 'react';
import renderer from 'react-test-renderer';
import FormHeader from './FormHeader';

describe('FormHeader', () => {
  it('Should render FormHeader', () => {
    const tree = renderer
      .create(
        <FormHeader
          label="label"
          title="title"
          controls={[{ label: 'button' }]}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
