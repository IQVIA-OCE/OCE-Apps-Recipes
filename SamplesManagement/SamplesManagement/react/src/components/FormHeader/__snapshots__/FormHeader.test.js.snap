// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHeader Should render FormHeader 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F6F6F6",
      "borderColor": "#DADADA",
      "borderWidth": 1,
      "flexDirection": "row",
      "padding": 12,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 3,
          "height": 35,
          "marginRight": 10,
          "width": 35,
        },
        Object {
          "backgroundColor": "#AEAEAE",
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 32,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï€„
    </Text>
  </View>
  <View
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#262626",
          "fontSize": 12,
          "letterSpacing": 2,
        }
      }
    >
      LABEL
    </Text>
    <Text
      style={
        Object {
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      title
    </Text>
  </View>
  <View />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingTop": 4,
      }
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#d9d9d9",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
          "elevation": 0,
          "marginLeft": 7,
          "minWidth": 64,
          "opacity": 1,
          "width": "auto",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onLayout={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "borderRadius": 4,
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "textAlign": "left",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    Object {
                      "color": "#0768FD",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "marginHorizontal": 16,
                      "marginVertical": 6,
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              button
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
